signature-envelope-cose.md

```diff
@@ -0,0 +1,134 @@
+
+```
+{
+  / crit / 2: [
+    / cty / 3,
+    'signingtime',
+    'expiry'
+  ],
+  / cty / 3: 'application/vnd.cncf.oras.artifact.descriptor.v1+json',
+  'signingtime': 1234567891000,
+  'expiry': 1234567891011
+}
+```
+
+Note: The above example is represented using the [extended CBOR diagnostic notation](https://datatracker.ietf.org/doc/html/rfc8152#appendix-C).
+
+- **`crit`** (*array of integers or strings*): This REQUIRED property (label: `2`) lists the headers that implementation MUST understand and process.
+  The array MUST contain `3` (`cty`), and `signingtime`. If `expiry` is presented, the array MUST also contain `expiry`.
+- **`cty`** (*string*): The REQUIRED property content-type (label: `3`) is used to declare the media type of the secured content (the payload).
+- **`signingtime`** (*datetime*): The REQUIRED property identifies the time at which the signature was generated.
+- **`expiry`** (*datetime*): This OPTIONAL property contains the expiration time on or after which the signature must not be considered valid.
+
+**UnprotectedHeaders**: Notary v2 supports two unprotected headers: `timestamp` and `x5chain`.
+
+```
+{
+  'timestamp': << TimeStampToken >>,
+  / x5chain / 33: [
+    << DER(leafCert) >>,
+    << DER(intermediate CACert) >>,
+    << DER(rootCert) >>
+  ]
+}
+```
+
+Note: `<<` and `>>` are used to notate the byte string resulting from encoding the data item.
+
+- **`timestamp`** (*byte string*): This OPTIONAL property is used to store time stamp token.
+  Only [RFC3161]([rfc3161](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2)) compliant TimeStampToken are supported.
+- **`x5chain`** (*array of byte strings*): This REQUIRED property (label: `33` by [IANA](https://www.iana.org/assignments/cose/cose.xhtml#header-parameters)) contains the list of X.509 certificate or certificate chain ([RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)) corresponding to the key used to digitally sign the COSE.
+  The certificate containing the public key corresponding to the key used to digitally sign the COSE MUST be the first certificate.
+  Optionally, this header can be presented in the protected header.
+
+**Signature**: In COSE, signature is calculated by constructing the `Sig_structure` for `COSE_Sign1`.
+The process is described below:
+
+1. Encode the protected header into a CBOR object as a byte string named `body_protected`.
+2. Construct the `Sig_structure` for `COSE_Sign1`.
+    ```
+    Sig_structure = [
+        / context / 'Signature1',
+        / body_protected / << ProtectedHeaders >>,
+        / external_aad / h'',
+        / payload / << Payload >>,
+    ]
+    ```
+3. Encode `Sig_structure` into a CBOR object as a byte string named `ToBeSigned`.
+4. Compute the signature on the `ToBeSigned` constructed in the previous step by using the signature algorithm defined in the corresponding header element: `alg`.
```

Comment:
`alg` is missing in the protected headers above.

Reply:
This is present in the spec